Issue with scipy 1.18.0

When using the newest version of scipy, I get the following error with sella. FYI @levineds, in case it impacts you. 

```python
/usr/share/miniconda/envs/quacc-env/lib/python3.12/site-packages/ase/optimize/optimize.py:344: in irun
    self.step()
/usr/share/miniconda/envs/quacc-env/lib/python3.12/site-packages/sella/optimize/optimize.py:240: in step
    s, smag = self._predict_step()
              ^^^^^^^^^^^^^^^^^^^^
/usr/share/miniconda/envs/quacc-env/lib/python3.12/site-packages/sella/optimize/optimize.py:229: in _predict_step
    s, smag = self.rs(
/usr/share/miniconda/envs/quacc-env/lib/python3.12/site-packages/sella/optimize/restricted_step.py:193: in __init__
    BaseRestrictedStep.__init__(self, pes, *args, **kwargs)
/usr/share/miniconda/envs/quacc-env/lib/python3.12/site-packages/sella/optimize/restricted_step.py:35: in __init__
    g = g0 + self.pes.get_H() @ self.scons
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/usr/share/miniconda/envs/quacc-env/lib/python3.12/site-packages/scipy/sparse/linalg/_interface.py:656: in __matmul__
    return self.__mul__(other)
           ^^^^^^^^^^^^^^^^^^^
/usr/share/miniconda/envs/quacc-env/lib/python3.12/site-packages/scipy/sparse/linalg/_interface.py:546: in __mul__
    return self.dot(x)
           ^^^^^^^^^^^
/usr/share/miniconda/envs/quacc-env/lib/python3.12/site-packages/scipy/sparse/linalg/_interface.py:616: in dot
    self._check_matching_namespace(x)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <48x48 ApproximateHessian with dtype=<class 'numpy.float64'>>
x = array([0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
       0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
       0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.])

    def _check_matching_namespace(self, x):
        xp_x = getattr(x, "_xp", None)
        if xp_x is None:
>           xp_x = array_namespace(x, self._xp.empty(0), sparse_ok=True)
                                      ^^^^^^^^
E           AttributeError: 'ApproximateHessian' object has no attribute '_xp'

/usr/share/miniconda/envs/quacc-env/lib/python3.12/site-packages/scipy/sparse/linalg/_interface.py:561: AttributeError
```

--- a/sella/linalg.py
+++ b/sella/linalg.py
@@ -31,7 +31,7 @@ class NumericalHessian(LinearOperator):
         else:
             n = self.ntrue
 
-        self.shape = (n, n)
+        super().__init__(self.dtype, (n, n))
 
         self.Vs = np.empty((self.ntrue, 0), dtype=self.dtype)
         self.AVs = np.empty((self.ntrue, 0), dtype=self.dtype)
@@ -109,8 +109,8 @@ class MatrixSum(LinearOperator):
         # For example, if two matrices are provided with the detypes
         # np.int64 and np.float64, then this MatrixSum object will be
         # np.float64.
-        self.dtype = sorted([mat.dtype for mat in matrices], reverse=True)[0]
-        self.shape = matrices[0].shape
+        dtype = sorted([mat.dtype for mat in matrices], reverse=True)[0]
+        super().__init__(dtype, matrices[0].shape)
 
         mnum = None
         self.matrices = []
@@ -153,8 +153,7 @@ class ApproximateHessian(LinearOperator):
         """A wrapper object for the approximate Hessian matrix."""
         self.dim = dim
         self.ncart = ncart
-        self.shape = (self.dim, self.dim)
-        self.dtype = np.float64
+        super().__init__(np.float64, (dim, dim))
         self.update_method = update_method
         self.symm = symm
         self.initialized = initialized
@@ -373,7 +372,7 @@ class SparseInternalJacobian(LinearOperator):
         self.indices = indices
         self.vals = vals
         self.nints = len(self.indices)
-        self.shape = (self.nints, 3 * self.natoms)
+        super().__init__(self.dtype, (self.nints, 3 * self.natoms))
 
     def asarray(self) -> np.ndarray:
         B = np.zeros((self.nints, self.natoms, 3))
@@ -418,7 +417,7 @@ class SparseInternalHessian(LinearOperator):
         vals: np.ndarray,
     ) -> None:
         self.natoms = natoms
-        self.shape = (3 * self.natoms, 3 * self.natoms)
+        super().__init__(self.dtype, (3 * natoms, 3 * natoms))
         self.indices = np.asarray(indices)
         self.vals = np.asarray(vals)
 

